### PR TITLE
[APPAI-1422] Resolve vulnerabilities for dependencies in Auth component

### PR DESCRIPTION
**Pull request by dgpatelgit**
Resolve vulnerabilities for dependencies in Auth component

Jira Issue :: https://issues.redhat.com/browse/APPAI-1422

Checklist ::
 - [X] Move any test specific packages from requirements.in to tests/requirements.in
 - [ ] (Not Applicable) Move any direct pip installations from Dockerfile.* to requirements.in
 - [X] Move any direct pip installations from tests/*.sh to tests/requirements.in
 - [X] Relative versioning in requirements.in & tests/requirements.in (no version pinning unless otherwise needed)
 - [X] Generate requirements.txt and tests/requirements.txt
 - [X] Update setup.py to use requirement.in

### CODE REVIEW

**Comment by arajkumar** [comment]
It would be great if you can add a check list in each PR. Something like below,

- [ ] Move any test specific packages from requirements.in to tests/requirements.in
- [ ] Move any direct pip installations from Dockerfile.* to requirements.in
- [ ] Move any direct pip installations from tests/*.sh to tests/requirements.in
- [ ] Relative versioning in requirements.in & tests/requirements.in (no version pinning unless otherwise needed)
- [ ] Generate requirements.txt and tests/requirements.txt
- [ ] Update setup.py to use requirement.in

**Comment by dgpatelgit** [comment]
> It would be great if you can add a check list in each PR. Something like below,
> 
> * [ ]  Move any test specific packages from requirements.in to tests/requirements.in
> * [ ]  Move any direct pip installations from Dockerfile.* to requirements.in
> * [ ]  Move any direct pip installations from tests/*.sh to tests/requirements.in
> * [ ]  Relative versioning in requirements.in & tests/requirements.in (no version pinning unless otherwise needed)
> * [ ]  Generate requirements.txt and tests/requirements.txt
> * [ ]  Update setup.py to use requirement.in (if setup.py exists)

Sounds good, added this check list to the PR.

**Comment by dgpatelgit** [comment]
> setup.py is not updated to use requirement.in.

This is done.

File: requirements.txt [hunk 1]
```diff
@@ -1,5 +1,34 @@
-flask
-cryptography
-flask-security
-pyjwt
-requests
\ No newline at end of file
+#
```
Comment (arajkumar) [inline requirements.txt:1]: BTW, who is really using the requirements.txt in the library. I think nobody.!

Comment (arajkumar) [reply requirements.txt:1]: If nobody uses then you can get rid of requirements.txt all together.

Comment (prashbnair) [reply requirements.txt:1]: Did not really understand this comment, isn't it required when a developer is working on this repo?

Comment (dgpatelgit) [reply requirements.txt:1]: We are using below imports in code:

import os
from functools import wraps

so we need requirments.txt right? @arajkumar can you elaborate your comment?

import jwt
import requests
from flask import current_app, request, g

Comment (arajkumar) [reply requirements.txt:1]: It is a library, the usual way of testing library is via unittest right? 
Who ever uses this library will get it's dependency via setup.py(requirement.in).

Comment (arajkumar) [reply requirements.txt:1]: and this library has only two impl files(auth.py, errors.py), rest all are supporting files.

**Comment by dgpatelgit** [comment]
@arajkumar requirements.txt is not used directly by this component, but when we include this component in say API server. It needs dependencies packages to be installed for 'Auth' component to function properly. This is my understanding, please correct me if I am missing something?

**Pull request merged**